codegenerator/cli/npm/envio/src/LazyLoader.res

```diff
@@ -1,95 +1,95 @@
 exception LoaderTimeout(string)
 
-type blockNumber = int
-
-let s = Belt.MutableSet.Int.make()
-
-// NOTE: any additional metadata that would be useful for debugging issues or displaying to users can be added here
-type asyncMapMetadata = {suggestedFix: string, asyncTaskName: string, caller: string}
-type asyncMap<'a> = {
+type rec asyncMap<'key, 'value> = {
   // The number of loaded results to keep cached. (For block loading this should be our maximum block interval)
   _cacheSize: int,
   // The maximum number of results we can try to load simultaneously.
   _loaderPoolSize: int,
   // How long to wait before retrying failures
-  // TODO: Add randomized exponential back-off
+  // TODO: Add randomized exponential back-off (outdated)
   _retryDelayMillis: int,
   // How long to wait before cancelling a load request
   _timeoutMillis: int,
   // The promises we return to callers. We satisfy them asynchronously.
-  externalPromises: dict<promise<'a>>,
+  externalPromises: dict<promise<'value>>,
   // The handled used to populate the external promises once we have loaded their data.
-  resolvers: dict<(. 'a) => unit>,
+  resolvers: dict<'value => unit>,
   // The keys currently being loaded
-  inProgress: Belt.MutableSet.Int.t,
+  inProgress: Utils.Set.t<'key>,
   // Keys  for items that we have not started loading yet.
-  loaderQueue: SDSL.PriorityQueue.t<int>,
+  loaderQueue: SDSL.Queue.t<'key>,
```

Comment:
I guess what makes me nervous of this polymorphic `'key` param, is that really it's constrained to a number or a string else there could be unexpected behaviours. If it were an object for instance then there is more nuance to using it.

My type OCD is making me want to see this `'key` type either statically typed as `Int(int) | Str(string)` or to pass a `toString` function rather than casting with Utils.magic